spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/BootApplicationConfigurationMetadataResolver.java

```diff
@@ -0,0 +1,211 @@
+
+	private static final String CONFIGURATION_METADATA_PATTERN = "classpath*:/META-INF/spring-configuration-metadata.json";
+
+	private static final String WHITELIST_PROPERTIES = "classpath*:/META-INF/spring-configuration-metadata-whitelist.properties";
+
+	public static final String CONFIGURATION_PROPERTIES_CLASSES = "configuration-properties.classes";
+
+	public static final String CONFIGURATION_PROPERTIES_NAMES = "configuration-properties.names";
+
+	private final Set<String> globalWhiteListedProperties = new HashSet<>();
+
+	private final Set<String> globalWhiteListedClasses = new HashSet<>();
+
+	private final ClassLoader parent;
+
+	public BootApplicationConfigurationMetadataResolver() {
+		this(null);
+	}
+
+	public BootApplicationConfigurationMetadataResolver(ClassLoader parent) {
+		this.parent = parent;
+		JarFile.registerUrlProtocolHandler();
+		try {
+			Resource[] globalResources = new PathMatchingResourcePatternResolver(ApplicationConfigurationMetadataResolver.class.getClassLoader()).getResources(WHITELIST_PROPERTIES);
+			loadWhiteLists(globalResources, globalWhiteListedClasses, globalWhiteListedProperties);
+		}
+		catch (IOException e) {
+			throw new RuntimeException("Error reading global white list of configuration properties", e);
+		}
+	}
+
+	@Override
+	public boolean supports(Resource app) {
+		File file = null;
+		try {
+			file = app.getFile();
+		}
+		catch (IOException e) {
+			return false;
+		}
+		Archive archive = null;
+		try {
+			archive = file.isDirectory() ? new ExplodedArchive(file) : new JarFileArchive(file);
+			return true;
+		}
+		catch (IOException | IllegalArgumentException e) {
+			return false;
+		}
+	}
+
+	/**
+	 * Return metadata about configuration properties that are documented via
+	 * <a href="http://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html">
+	 * Spring Boot configuration metadata</a> and visible in an app.
+	 * @param app a Spring Cloud Stream app; typically a Boot uberjar,
+	 *            but directories are supported as well
+	 */
+	public List<ConfigurationMetadataProperty> listProperties(Resource app, boolean exhaustive) {
+		File moduleFile = null;
+		try {
+			moduleFile = app.getFile();
+		}
+		catch (IOException e) {
+			return Collections.emptyList();
+		}
+		Archive archive = null;
+		try {
+			archive = moduleFile.isDirectory() ? new ExplodedArchive(moduleFile) : new JarFileArchive(moduleFile);
+			return listProperties(archive, exhaustive);
+		}
+		catch (IOException e) {
+			throw new RuntimeException("");
+		}
+	}
+
+	public List<ConfigurationMetadataProperty> listProperties(Archive archive, boolean exhaustive) {
+		ClassLoader moduleClassLoader = null;
+		try {
+
+			List<ConfigurationMetadataProperty> result = new ArrayList<>();
+			moduleClassLoader = new BootClassLoaderCreation(archive, parent).createClassLoader();
+			ResourcePatternResolver moduleResourceLoader = new PathMatchingResourcePatternResolver(moduleClassLoader);
+
```

Comment:
Are these extra lines for readability? I think they can be avoided and probably move into some private methods if needed.